test/util/stream_utils_unit.js

```diff
@@ -498,4 +498,153 @@ describe('StreamUtils', () => {
       expect(manifest.textStreams[1].id).toBe(2);
     });
   });
+
+  describe('filterVariantsBySize_', () => {
+    const avc1Variant1080 = /** @type {shaka.extern.Variant} */(
+      /** @type {?} */({
```

Comment:
This is a bit problematic, since you're disabling type-checking on the inside of this expression.  If we change the definition of Variant in some way that breaks this, we might not notice.

This should get replaced by a builder such as shaka.test.ManifestGenerator.Variant.  Reviewing that, though, it appears that it needs some cleanup, too, to make it usable directly instead of through shaka.test.ManifestGenerator.

Can you please add a TODO comment about it, and we can work on that after your PR is merged?